airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -148,7 +149,21 @@ def __init__(
             except ValueError:
                 raise VaultError(f"The version is not an int: {conn_version}. ")
 
-        if auth_type in ["approle", "aws_iam"]:
+        if auth_type == "approle":
+            if not role_id:
+                if self.connection.extra_dejson.get('role_id'):
+                    role_id = self.connection.extra_dejson.get('role_id')
```

Comment:
Almost perfect :).

I think we not only want to warn when uses passes role_id for "approle" but we also want to deprecate this (pseudo-code):

```
dummy_login:password@host?auth_type="approle"&role_id="nn"
```

we should here raise a similar warning as below and ask users to change it to:

```
role:password@host?auth_type="approle"
```

Reply:
@potiuk Yeah, fair enough.  What about?

```
if auth_type == "approle":
    if role_id:
        warnings.warn(
            """The usage of role_id for AppRole authentication has been deprecated.
            Please use connection login.""",
            DeprecationWarning,
            stacklevel=2,
        )
    if self.connection.extra_dejson.get('role_id'):
        role_id = self.connection.extra_dejson.get('role_id')
        warnings.warn(
            """The usage of role_id in connection extra for AppRole authentication has been deprecated.
            Please use connection login.""",
            DeprecationWarning,
            stacklevel=2,
        )
    elif self.connection.login:
        role_id = self.connection.login
```

Happy to rephrase the messages as you suggest.

Reply:
I think this provides different warnings for the current ways of submitting `role_id` without breaking anything.

Reply:
I'd say this (if-> elif):

```
if auth_type == "approle":
    if role_id:
        warnings.warn(
            """The usage of role_id for AppRole authentication has been deprecated.
            Please use connection login.""",
            DeprecationWarning,
            stacklevel=2,
        )
    elif self.connection.extra_dejson.get('role_id'):
        role_id = self.connection.extra_dejson.get('role_id')
        warnings.warn(
            """The usage of role_id in connection extra for AppRole authentication has been deprecated.
            Please use connection login.""",
            DeprecationWarning,
            stacklevel=2,
        )
    elif self.connection.login:
        role_id = self.connection.login
```
otherwise when you pass both `role_id` as parameter and login, the login one will be used, which is unexpected.

Reply:
Good catch!